main.go

```diff
@@ -254,95 +310,131 @@ func _main() error {
 	if err != nil {
 		return err
 	}
+	projectID := projectTmp.Name()
+
+	if cache != nil {
+		stdinTmpCopy := false
+		projectID, stdinTmpCopy, err = getFilesContentHash(files, projectTmp.Name())
+		if err != nil {
+			log.Printf("Error creating hash for cache mode: %v, defaulting to normal mode", err)
+			*cache = false
+		}
+		if stdinTmpCopy {
+			files = append(files, projectTmp.Name())
+		}
+	}
 	defer os.Remove(projectTmp.Name())
+
 	project := &runner.ProjectState{
-		Id: projectTmp.Name(),
+		Id: projectID,
 		Pages: []runner.ProjectPage{
 			{
 				Panels: nil,
 			},
 		},
 	}
 
+	settings := *runner.DefaultSettings
 	tmpDir, err := os.MkdirTemp("", "dsq")
 	if err != nil {
 		return err
 	}
 	defer os.RemoveAll(tmpDir)
-
-	ec := runner.NewEvalContext(*runner.DefaultSettings, tmpDir)
-	for i := 0; i < len(files); i++ {
-		file := files[i]
-		panelId := uuid.New().String()
-		resultFile := ec.GetPanelResultsFile(project.Id, panelId)
-		out, err := openTruncate(resultFile)
-		if err != nil {
-			return err
+	if cache != nil {
+		tmpDir = "dsq-cache"
+		settings.CacheMode = cache
+		if info, err := os.Stat(getCachedDBPath(projectID)); err != nil || info.Size() == 0 {
+			log.Println("SQLite file is not found on disk, creating a new one...")
+			*cache = false
 		}
+	}
 
-		readFromStdin := false
-		if file == "" {
-			b, err := ioutil.ReadAll(os.Stdin)
-			if err == nil {
-				if i == len(files)-1 {
-					return errors.New("Expected file extension or mimetype: e.g. cat x.csv | dsq -s csv, or cat x.csv | dsq -s text/csv")
-				}
-				mimetype := files[i+1]
-				if !strings.Contains(mimetype, string(filepath.Separator)) {
-					mimetype = string(runner.GetMimeType("x."+mimetype, runner.ContentTypeInfo{}))
-				}
-
-				if mimetype == "" {
-					return fmt.Errorf("Unknown mimetype or file extension: %s.", mimetype)
+	ec := runner.NewEvalContext(settings, tmpDir)
+	if cache == nil || *cache == false {
+		for i := 0; i < len(files); i++ {
```

Comment:
Getting pretty nested inside here. Let's move this into a new function?

```
if cache == nil || *cache == false {
  importFiles(files)
}
```

Reply:
sure!